autopilot.go

```diff
@@ -23,39 +25,69 @@ func main() {
 
 type AutopilotPlugin struct{}
 
+func venerableAppName(appName string) string {
+	return fmt.Sprintf("%s-venerable", appName)
+}
+
+func getActionsForExistingApp(appRepo *ApplicationRepo, appName, manifestPath, appPath string) []rewind.Action {
+	return []rewind.Action{
+		// rename
+		{
+			Forward: func() error {
+				return appRepo.RenameApplication(appName, venerableAppName(appName))
+			},
+		},
+		// push
+		{
+			Forward: func() error {
+				return appRepo.PushApplication(manifestPath, appPath)
+			},
+			ReversePrevious: func() error {
+				// If the app cannot start we'll have a lingering application
+				// We delete this application so that the rename can succeed
+				appRepo.DeleteApplication(appName)
```

Comment:
@xoebus
@dprotaso and I debated returning the error from DeleteApplication if any occurs, but there's no guarantee that when the push failed the app was created at all. In that case we'd be returning a lot of failures to delete the non-existent app, and not continue on to the rename.

If for some reason we couldn't delete the application (but it existed) then the following rename attempt would error out.